specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/stable/2017-11-01/notificationhubs.json

```diff
@@ -0,0 +1,1971 @@
+                "$ref": "#/definitions/PolicykeyResource"
+              },
+              "description": "Parameters supplied to regenerate the NotificationHub Authorization Rule Key."
+            },
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            },
+            {
+              "$ref": "#/parameters/SubscriptionIdParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "Regenerate the Primary/Secondary Key and returns the Keys and the connection string",
+              "schema": {
+                "$ref": "#/definitions/ResourceListKeys"
+              }
+            }
+          }
+        }
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NotificationHubs/namespaces/{namespaceName}/notificationHubs/{notificationHubName}/pnsCredentials": {
+        "post": {
+          "tags": [
+            "NotificationHubs"
+          ],
+          "operationId": "NotificationHubs_GetPnsCredentials",
+          "description": "Lists the PNS Credentials associated with a notification hub .",
+          "parameters": [
+            {
+              "name": "resourceGroupName",
+              "in": "path",
+              "required": true,
+              "type": "string",
+              "description": "The name of the resource group."
+            },
+            {
+              "name": "namespaceName",
+              "in": "path",
+              "required": true,
+              "type": "string",
+              "description": "The namespace name."
+            },
+            {
+              "name": "notificationHubName",
+              "in": "path",
+              "required": true,
+              "type": "string",
+              "description": "The notification hub name."
+            },
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            },
+            {
+              "$ref": "#/parameters/SubscriptionIdParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "Gets all the NotificationHub PNS credentials",
+              "schema": {
+                "$ref": "#/definitions/PnsCredentialsResource"
+              }
+            }
+          }
+        }
+      }
+    },
+    "definitions": {
+      "CheckNameAvailabilityRequestParameters": {
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "Resource name"
+          },
+          "type": {
+            "readOnly": true,
+            "type": "string",
+            "description": "Resource type"
+          }
+        },
+        "required": [
+          "name"
+        ],
+        "description": "Parameters supplied to the Check Name Availability for Namespace and NotificationHubs."
+      },
+      "CheckNameAvailabilityResponse": {
+        "properties": {
+          "nameAvailable": {
```

Comment:
Are you 100% this is the **exact** JSON key used? Previous API version was cased "NameAvailable".
Note that C# JSON parser is case-insensitive (and that's a problem), so even if this Swagger is working in C#, does not mean this is correct. We need to be sure this is exactly what Azure is returning.
Thanks,

Reply:
This is what the azure validator warning mentions....to make it camelcase. Also this is a breaking change so changing the casing shouldnt matter right?

Reply:
@amolr There is two things:
- The validator validates against **convention**. It doesn't know what the RestAPI **actually** returns.
- The Swagger **MUST** tell the truth of the RestAPI. 

If the RestAPI does not follow the convention, you will get a warning **and you can't fix it**. You might add a "warning suppression" in the Readme to avoid this message.
Changing the case in the Swagger to remove the warnings is like breaking you car dashboard to turn off the "engine check" light: this does not address the real issue.

@salameer This is the second time I need to explain the difference between validation and RestAPI accurancy, I think there is a miss of documentation to help people understand what is the right fix.

Reply:
@amolr -- if it's actually uppercase (ie `NameAvailable`) on the wire, we can add a suppression for error. @lmazuel  is correct, the swagger must be accurate .